### PR TITLE
2. UI Tests

### PR DESCRIPTION
**Pull request by diarmaidlindsay**
### Issue

- N/A

### Motivation and Context

- Write Sample UI Tests which demonstrate mocking of the Repository and API Client, and verification of UI Element display.

### Modified points

- Added UI Tests.

### Description and Screenshots

image

### Figma or screen image URL

- N/A

### Worries or Issues

- Would like to mock goRouter to be able to write more complex UI tests which go to different screens.

### CODE REVIEW

**Review by dinkar1708 — COMMENTED** [review]
better to have

pr description a litt
read me section about document follow for test case
how to run test case - command is better if can

File: test/presentation/feature/news/top_anime_page_test.dart [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import 'package:flutter_template_riverpod/data/repository/my_anime_list_repository.dart';
+import 'package:flutter_template_riverpod/presentation/feature/news/top_anime_page.dart';
+import 'package:flutter_template_riverpod/presentation/feature/news/widgets/top_anime_item_widget.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'top_anime_page_test.mocks.dart';
+
+Finder _gridItemText(int id) =>
```
Comment (dinkar1708) [inline test/presentation/feature/news/top_anime_page_test.dart]: better name can be
_findGridItemTitleText

**Review by dinkar1708 — APPROVED** [review]
optional comment
lgtm

File: test/presentation/feature/news/top_anime_page_test.dart (hunk 1 above)
Comment (diarmaidlindsay) [reply test/presentation/feature/news/top_anime_page_test.dart]: Thank you, I corrected it.

**Pull request merged**